lib/configurator.js

```diff
@@ -27,6 +28,19 @@ function validateOptions (options) {
   )
 }
 
+var defaultCookieDomain = (function () {
+  var domainChunks = document.location.hostname.split('.')
+  for (var i = domainChunks.length - 2; i >= 1; i--) {
+    var domain = domainChunks.slice(i).join('.')
+    var name = '_td_domain_' + domain // append domain name to avoid race condition
+    cookie.setItem(name, domain, 3600, '/', domain)
+    if (cookie.getItem(name) === domain) {
+      return domain
+    }
+  }
+  return document.location.hostname
+})()
```

Comment:
e106b4c983b59fc618cb849331fd78e117da2c06 made it possible to use a function for `cookieDomain` but here still evaluates the function immediately. Is this intentional? If so, Please go ahead.

My suggestion on Slack was using a definition like following code so that actual evaluation of `evaluateDefaultCookieDomain` is delayed until the cookie domain is actually used (which is at `fetchServerCookie`) and it's not evaluated twice when `cookieDomainHost(this._serverCookieDomain)` is called.

```
var defaultCookieDomain = function() {
  if (!('result' in defaultCookieDomain)) {
      // evaluateDefaultCookieDomain is equivalent to current defaultCookieDomain function
      defaultCookieDomain.result = evaluateDefaultCookieDomain();
  }
  return defaultCookieDomain.result
}
```

or to be slightly safer:

```
var defaultCookieDomainCache = function() {
  if (!('result' in defaultCookieDomainCache)) {
      defaultCookieDomainCache.result = evaluateDefaultCookieDomain();
  }
  return defaultCookieDomainCache.result
}
var defaultCookieDomain = defaultCookieDomainCache
```